src/compiler/checker.ts

```diff
@@ -3112,6 +3112,9 @@ namespace ts {
                         return "(Anonymous function)";
                 }
             }
+            if (hasProperty(symbol, "syntheticName")) {

```

Comment:
`(symbol as TransientSymbol).syntheticName` is falsy for `""`, so it would have to be `typeof (symbol as TransientSymbol).syntheticName === "string"`